### ISSUE TITLE
estimate_mix_props should drop all-zero columns

### ISSUE THREAD

**Issue by willwerscheid** [stephens999/ashr#89]
It's possible that we get columns of all zeros when calculating likelihood matrices (and thus -Inf in llik matrices). In the example that's throwing an error for me, the smallest mixture component (in a mixture of uniforms) is `[0, 5.80e-05]`, but the data points `x` are all around `0.15` with small standard errors, so `log_comp_dens_conv.unimix` evaluates the likelihood that they come from this component as numerically zero. So we get a column of all zeros in the likelihood matrix. This causes `mixsqp` to complain. We could either drop such columns in `ashr:::estimate_mix_prop` or have `mixsqp` deal with them appropriately. I think I prefer the former option.

**Comment by stephens999** [comment]
In principle I think mixsqp should deal with -inf

**Comment by stephens999** [comment]
Or zeros

**Comment by stephens999** [comment]
@willwerscheid Can you open an issue in mixsqp?

**Comment by willwerscheid** [comment]
Ok, moving issue to `mixsqp` and closing.

**Comment by willwerscheid** [comment]
Reopening because this causes problems for mixIP as well. I get the following warning:
```Warning in mixIP(matrix_lik = c(0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,  :
  Optimization step yields mixture weights that are either too small, or negative; weights have been corrected and renormalized after the optimization.```
I think we should fix this here rather than (or in addition to) `mixSQP`.

**Comment by willwerscheid** [comment]
I am now thinking that we ought to improve the grid selection. We usually get columns of zeros because `autoselect.mixsd` adds very small sds (down to (`min(s) / 10`)), which doesn't work so well if say your observations are all bounded away from zero with small SEs. An easy fix would be to set `sigmamin` no smaller than `sqrt(min(x^2 - s^2))/C` (with C > 1 some constant).
This should clear up most cases that we are seeing now, but won't solve the problem if the user passes in a `g` with a bad grid. So I do still think we need to check for columns of zeros in `estimate_mixprop`.

**Comment by stephens999** [comment]
i'm not sure that will work.

if x=0 and s=1 your formula gets sigmamin = 1/C
whereas in that can you want  a small sigmamin i think

**Comment by willwerscheid** [comment]
1/C is what you get now (with C = 10). My formula gives -1/C. What do you propose?

**Comment by willwerscheid** [comment]
I was just planning to use the max of sigmamin as it is currently calculated and the formula above. But I think you make a good argument for changing the former.

**Comment by willwerscheid** [comment]
@stephens999 
Maybe something like:
```
  tmp = betahat^2 - sebetahat^2
  sigmamax = mult^2 * sqrt(max(c(0, tmp)))
  if (sigmamax == 0) {
    # Everything is null.
    return(sqrt(.Machine$double.eps))
  }
  sigmamin = sqrt(min(tmp[tmp > 0])) / mult^2
```

**Comment by willwerscheid** [comment]
This was fixed by PR #93
